src/averageAges.js

```diff
@@ -55,7 +73,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child =>
+    onlyWithSon
+      ? child.sex === 'm' && people.some(mother => mother.name === child.mother)
+      : people.some(mother => mother.name === child.mother));
+
+  const motherAgeAtBirthChild = children.map(child => {
+    const mom = people.find(person => person.name === child.mother);
+
+    return child.born - mom.born;
+  });
+
+  const sum = motherAgeAtBirthChild.reduce((prev, current) =>
+    prev + current, 0);
+  const avgAge = sum / motherAgeAtBirthChild.length;
+
+  return avgAge;
```

Comment:
I see you lerarn a lot of methods
 Its great, but i want to show example, how to improve performance and avoid a lot of iterations the same array. 
```
 const mothers = people.reduce((acc, person) => {
    acc[person.name] = person;

    return acc;
  }, {});

  const { sum, childrenCount } = people.reduce((acc, person) => {
    const { sex, mother, born } = person;

    const isPerson = onlyWithSon
      ? sex === 'm' && mothers[mother]
      : mothers[mother];

    return isPerson ? {
      sum: acc.sum + born - mothers[mother].born,
      childrenCount: acc.childrenCount + 1,
    } : acc;
  }, {
    sum: 0, childrenCount: 0,
  });

  return sum / childrenCount;